stores/nvidia.py

```diff
@@ -456,13 +461,16 @@ def add_to_cart(self, product_id):
                         if error["code"] == "invalid-product-id":
                             raise ProductIDChangedException()
                 except json.decoder.JSONDecodeError as er:
-                    log.warning(f"Failed to decode json: {response.text}")
+                    log.warning(f"Failed to decode json: {response.text}, due to {er}")
             else:
                 log.debug("item not in stock")
                 return False
-        except Exception as ex:
-            log.warning(str(ex))
-            log.warning("The connection has been reset.")
+        except (
+            ConnectionResetError,
+            requests.exceptions.ConnectionError,
+            requests.exceptions.Timeout,  # covers ConnectTimeout + ReadTimeout
+        ) as ex:
+            log.warning(f"The connection experienced a problem: {ex}.")
```

Comment:
base Exception already catches all of these I believe.

Reply:
Yup, it does :)